src/test/java/com/example/ajaxonboardingproject/CinemaHallRepositoryTests.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.ajaxonboardingproject
+
+import com.example.ajaxonboardingproject.model.CinemaHall
+import com.example.ajaxonboardingproject.repository.CinemaHallRepository
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.test.StepVerifier
+
+@SpringBootTest
+class CinemaHallRepositoryTests {
+    @Autowired
+    private lateinit var cinemaHallRepository: CinemaHallRepository
+
+    @BeforeEach
+    fun clearCollection() {
+        cinemaHallRepository.deleteAll().subscribe()
+    }
+
+    @Test
+    fun saveCinemaHallToDbTestOk() {
+        //Given
+        val cinemaHall = CinemaHall(capacity = 100, description = "grate one")
+        //When
+        val actual: Mono<CinemaHall> = cinemaHallRepository.save(cinemaHall)
+        //Then
+        val expected = actual.flatMap {
+            cinemaHallRepository.findById(it.id)
```

Comment:
let's not modify the actual chain. Instead, make validation that save method returned entity with same parameters as passed

If you want to assert that the entity is actually saved, use save().block() and in StepVerifier use findById

In your test you're saving same entity twice currently

Reply:
Done!